sktm/jenkins.py

```diff
@@ -401,23 +438,16 @@ def _params_eq(self, build, params):
 
     def find_build(self, params, eid=None):
         job = self.__get_job()
-        lbuild = None
-
-        while not lbuild:
-            try:
-                lbuild = job.get_last_build()
-            except jenkinsapi.custom_exceptions.NoBuildData:
-                time.sleep(1)
+        lbuild = self.__get_last_build(job, 1)
 
         if eid is not None:
             while lbuild.get_number() < eid:
-                time.sleep(1)
```

Comment:
I wouldn't remove the sleep here since it serves a different purpose than retry interval - we expect the retrieval itself to complete, but want to wait before doing it again.

Reply:
OK. time.sleep(1) is added back.